.github/workflows/build_test_windows.yaml

```diff
@@ -11,13 +11,23 @@ jobs:
     runs-on: windows-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v1
+      uses: actions/checkout@v4
 
-    - id: build
+    - name: Create build folder
       run: |
         cd ${{ github.workspace }}
         mkdir build
+
+    - name: Setup build directory cache
+      uses: actions/cache@v4
+      with:
+        path: ${{ github.workspace }}/build
+        key: build-windows-cache-${{ github.ref }}
+
+    - name: Build DPsim
+      run: |
+        cd ${{ github.workspace }}
         cd build
         git describe --tags --abbrev=0 --match "v*"
         cmake -DWITH_PYBIND=OFF ..
-        cmake --build . --target dpsim
+        cmake --build . --target dpsim --parallel
```

Comment:
Well, in theory it should do it automatically, but there are mixed opinions about the actual performance across the internet for the case of a github windows runner.
We could use something like `NumberOfLogicalProcessors` for consistency, because `nproc` is in theory not implemented in `pwsh` or use the action `SimenB/github-actions-cpu-cores@v2`